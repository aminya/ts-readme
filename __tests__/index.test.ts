import { getAllDocs, generateMarkdown } from '../src/index.js';
import path from 'path';

const __dirname = path.dirname(new URL(import.meta.url).pathname);

test('it should generate some simple types', async () => {
  const docs = await getAllDocs(path.join(__dirname, '__fixtures__/simple.ts'));
  expect(docs.map((doc) => generateMarkdown(doc))).toMatchSnapshot();
});
