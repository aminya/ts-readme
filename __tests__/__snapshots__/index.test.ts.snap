// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should generate some simple types 1`] = `
[
  "### \`bar\` (variable)

",
  "### \`foo\` (variable: number)

",
  "### \`Interface\` (interface)

**Members:**

- one (\`boolean\`)
- two (\`number\`)

",
  "### \`Type\` (type)

",
  "### \`add\` (function)

Add two numbers

**Parameters:**

- left (\`number\`) - the first number
- right (\`number\`) - the second number

**returns:** string

\`\`\`tsx
add(1, 2)
\`\`\`


",
  "### \`subtract\` (function)

Subtract two numbers

**Parameters:**

- left (\`number\`) 
- right (\`any\`) 

**returns:** string

",
  "### \`withCallback\` (function)

**Parameters:**

- foo (\`(value: string) => string\`) - the callback

**returns:** string

",
  "### \`Car\` (class)

A car

**Properties:**

- wheels - A circle thing

",
]
`;
